src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidContentProviderBroker.cs

```diff
@@ -0,0 +1,349 @@
+            string className = bundleResult.GetString("broker.activity.name");
+            string uid = bundleResult.GetString("caller.info.uid");
+
+            Intent brokerIntent = new Intent();
+            brokerIntent.SetPackage(packageName);
+            brokerIntent.SetClassName(
+                    packageName,
+                    className
+            );
+
+            brokerIntent.PutExtras(bundleResult);
+            brokerIntent.PutExtra(BrokerConstants.NegotiatedBPVersionKey, _brokerHelper.NegotiatedBrokerProtocalKey);
+
+            var interactiveIntent = brokerIntent.PutExtras(_brokerHelper.GetInteractiveBrokerBundle(brokerRequest));
+            interactiveIntent.PutExtra(BrokerConstants.CallerInfoUID, Binder.CallingUid);
+
+            return interactiveIntent;
+        }
+
+        public async Task<MsalTokenResponse> AcquireTokenSilentAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters acquireTokenSilentParameters)
+        {
+            _brokerHelper.CheckPowerOptimizationStatus();
+
+            InitiateBrokerHandshakeAsync();
+
+            BrokerRequest brokerRequest = BrokerRequest.FromSilentParameters(
+                authenticationRequestParameters, acquireTokenSilentParameters);
+
+            return await PerformAcquireTokenSilentAsync(brokerRequest).ConfigureAwait(false);
+        }
+
+        private async Task<MsalTokenResponse> PerformAcquireTokenSilentAsync(BrokerRequest brokerRequest)
+        {
+            try
+            {
+                return await AcquireTokenSilentViaBrokerAsync(brokerRequest).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorPiiWithPrefix(ex, "Android broker silent invocation failed. ");
+                _brokerHelper.HandleBrokerOperationError(ex);
+                throw;
+            }
+        }
+
+        private async Task<MsalTokenResponse> AcquireTokenSilentViaBrokerAsync(BrokerRequest brokerRequest)
+        {
+            using (_logger.LogMethodDuration())
+            {
+                _logger.Verbose("User is specified for silent token request. Starting silent Android broker request. ");
+
+                brokerRequest = await UpdateRequestWithAccountInfoAsync(brokerRequest).ConfigureAwait(false);
+
+                string silentResult = await PerformAcquireTokenSilentFromBrokerAsync(brokerRequest).ConfigureAwait(false);
+                return _brokerHelper.HandleSilentAuthenticationResult(silentResult, brokerRequest.CorrelationId);
+            }
+        }
+
+        /// <summary>
+        /// This method is only used for Silent authentication requests so that we can check to see if an account exists in the account manager before
+        /// sending the silent request to the broker. 
+        /// </summary>
+        public async Task<BrokerRequest> UpdateRequestWithAccountInfoAsync(BrokerRequest brokerRequest)
```

Comment:
not public, also probably not needed.

Reply:
So I did this and named it AndroidBrokerStaticHelper. is that name ok?